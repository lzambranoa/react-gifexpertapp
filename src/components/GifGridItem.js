import React from 'react';

export const GifGridItem = ( {id, title, url} ) => {

    console.log(id, title, url);
    return (
        <div className='card animate__animated animate__fadeInDown animate__delay-2s'>
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}