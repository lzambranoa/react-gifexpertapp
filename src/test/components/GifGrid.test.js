import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import '@testing-library/jest-dom';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs'); // Finge la llamada a este archivo


describe('Pruebas al componente GifGrid', () => {

    const category = 'One Punch';
    

    test('Debe mostrar correctamente el GifGrid', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category={category}/>);
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se carga images UseFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localohost/prueba',
            title: 'Leonardo'
        },
        {
            id: 'DEF',
            url: 'https://localohost/prueba',
            title: 'Leonardo'
        },
    ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>); 

        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('p').exists()).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe(gifs.length);
    });
});