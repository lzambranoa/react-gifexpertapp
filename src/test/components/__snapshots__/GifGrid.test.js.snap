// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifGrid Debe mostrar correctamente el GifGrid 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas al componente GifGrid debe de mostrar items cuando se carga images UseFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Leonardo"
      url="https://localohost/prueba"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="Leonardo"
      url="https://localohost/prueba"
    />
  </div>
</Fragment>
`;
